### PR TITLE
improve: add all chains and fix issues for tx history page

### PR DESCRIPTION
**Pull request by amateima**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ This project was bootstrapped with [Create React App](https://github.com/faceboo
 - REACT_APP_DISABLE_DEPOSITS: Displays a maintenance banner and disables the AddLiquidityForm in the Pool view.
 - REACT_APP_ENABLE_REACT_QUERY_DEV_TOOLS: Display React-Query dev tools if set to `true`.
 - REACT_APP_MATOMO_URL: Enable matomo by setting the matomo url, app will ignore events if not defined.
+- REACT_APP_CHAIN_137_PROVIDER_URL: custom provider for Polygon node
+- REACT_APP_CHAIN_42161_PROVIDER_URL: custom provider for Arbitrum node
```
Comment (amateima) [inline README.md:19]: Add separate providers for Arbitrum and Polygon. Infura's performance is really bad on these chains while fetching events. They require a 5k block range

File: src/state/transferHistory.ts [hunk 2]
```diff
@@ -31,17 +31,18 @@ export function createTxHistoryClientConfig(params: Params): {
...
       chainId,
       spokeAddress: config.getSpokePoolAddress(chainId),
     };
   });
-  client = new TransfersHistoryClient(createTxHistoryClientConfig(params));
+  const client = new TransfersHistoryClient(
+    createTxHistoryClientConfig(params)
+  );
+  // Uncomment this only for debugging
+  // client.setLogLevel("debug");
```
Comment (amateima) [inline src/state/transferHistory.ts:46]: I don't want the tx history client to be cached. It should be instantiated every time the tx history page is mounted and cleared when the tx history page is unmounted.

File: src/utils/constants.ts [hunk 3]
```diff
@@ -454,6 +455,12 @@ export const matomoUrl = process.env.REACT_APP_MATOMO_URL;
 export const MAX_APPROVAL_AMOUNT = ethers.constants.MaxUint256;
 export const FEE_ESTIMATION = ".004";
 export const AddressZero = ethers.constants.AddressZero;
+export const ArbitrumProviderUrl =
+  process.env.REACT_APP_CHAIN_42161_PROVIDER_URL ||
+  `https://arbitrum-mainnet.infura.io/v3/${infuraId}`;
+export const PolygonProviderUrl =
+  process.env.REACT_APP_CHAIN_137_PROVIDER_URL ||
+  `https://polygon-mainnet.infura.io/v3/${infuraId}`;
```
Comment (amateima) [inline src/utils/constants.ts:463]: Use custom ENV variables for Arbitrum and Polygon providers

File: src/views/Transactions/useTransactionsView.ts [hunk 4]
```diff
@@ -5,9 +5,8 @@ import useWindowSize from "hooks/useWindowsSize";
 import getTxClient from "state/transferHistory";
 import { Transfer } from "@across-protocol/sdk-v2/dist/transfers-history/model";
 import { transfersHistory } from "@across-protocol/sdk-v2";
-import { usePrevious } from "hooks";
```
Comment (amateima) [inline src/views/Transactions/useTransactionsView.ts:8]: Dont need to track the previous value using this hook. The value of the previous account can be easily tracked when the cleanup function of the useEffect runs

File: src/views/Transactions/useTransactionsView.ts [hunk 5]
```diff
@@ -17,53 +16,46 @@ export default function useTransactionsView() {
...
-    if (account && initialLoading) {
-      txHistoryClient.startFetchingTransfers(account).catch((err) => {
-        console.error(
-          "Error in startFetchingTransfers call in txHistoryClient",
-          err
-        );
-      });
-      txHistoryClient.on(
+    if (txClient) {
+      txClient.on(
```
Comment (amateima) [inline src/views/Transactions/useTransactionsView.ts:25]: Attach the event listener when the component (page) is mounted

File: src/views/Transactions/useTransactionsView.ts [hunk 6]
```diff
@@ -17,53 +16,46 @@ export default function useTransactionsView() {
...
-    previousAccount,
-    txHistoryClient,
-  ]);
+
+    return () => {
+      if (account && txClient) {
+        txClient.stopFetchingTransfers(account);
+        setRawFilledTx([]);
+        setRawOngoingTx([]);
+      }
```
Comment (amateima) [inline src/views/Transactions/useTransactionsView.ts:56]: Clean up the state before the account is changed or page unmounts

**Review by Tulun — APPROVED** [review]
LGTM

File: src/views/Transactions/useTransactionsView.ts (hunk 6 above)
Comment (mrice32) [reply src/views/Transactions/useTransactionsView.ts:56]: +1

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**